components/patient-access-rules/api/src/main/java/org/phenotips/data/permissions/script/RecordLockingService.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.permissions.script;
+
+import org.phenotips.Constants;
+import org.phenotips.data.Patient;
+import org.phenotips.data.PatientRepository;
+import org.phenotips.data.permissions.PatientAccess;
+import org.phenotips.data.permissions.PermissionsManager;
+import org.phenotips.data.permissions.internal.access.ManageAccessLevel;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.context.Execution;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.script.service.ScriptService;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * A service to add or remove a lock object to a record. The lock object will remove edit rights for all users on the
+ * document.
+ * @version $Id$
+ * @since 1.3
+ */
+@Component
+@Named("recordLocking")
+@Singleton
+public class RecordLockingService implements ScriptService
```

Comment:
No problem.  But regarding your comment above about the script service having two methods and taking a patient object. If I implement that then there will need to be velocity code which processes the string representation of the patient before calling the script service. Basically there will just be more velocity code. Wouldn't it be better to have the script service deal with the raw request parameter values rather than putting that logic in velocity?
The component will have two methods.